client/src/app/users/user-list.component.ts

```diff
@@ -12,25 +12,23 @@ import { Subscription } from 'rxjs';
 
 export class UserListComponent implements OnInit, OnDestroy  {
   // These are public so that tests can reference them (.spec.ts)
-  public serverFilteredUsers: User[];
-  public filteredUsers: User[];
+  public serverFilteredUsers: User[] = [];
+  public filteredUsers: User[] = [];
 
-  public userName: string;
-  public userAge: number;
-  public userRole: UserRole;
-  public userCompany: string;
+  public userName: string = '';
+  public userAge: number = -1;
+  public userRole: UserRole = 'viewer';
+  public userCompany: string = '';
```

Comment:
I don't think we want to set defaults for the filters like this.

Reply:
I didn't like that either, but I wasn't sure how to avoid it. These essentially become the "default" values of the filter fields in the user list component.

A lot of these changes were really rushed. I initially thought it would be "easy" to fix these things and I was trying to include it in the Angular 10 update (#136). After a while, though, I realized that I was typing more than thinking and decided to split this off, so we could look it over and have exactly this sort of conversation.

So what do you think would be a better approach? Make them all optional? That seems to be an option, although I'm nervous about just making a bazillion things optional as it seems sloppy/lazy to me.

Reply:
Is there something preventing them from just being uninitialized?

Reply:
Yup. With all the strictness stuff turned on you get errors because it can't be sure that they have values when they're used. This may indicate a weakness in the overall design of this component?